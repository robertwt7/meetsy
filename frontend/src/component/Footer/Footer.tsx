import { FunctionComponent } from "react";
import { Route } from "src/config/router";
import { NavigationList, INavigationItemProps } from "../Navigation";
import { Copyright } from "../Copyright";

export const Footer: FunctionComponent = () => {
  const items: INavigationItemProps[] = [
    {
      text: "Privacy Policy",
      testId: "privacypolicy",
      route: Route.PRIVACY_POLICY,
    },
    {
      text: "Terms of Service",
      testId: "termsOfService",
      route: Route.TERMS_OF_SERVICE,
    },
  ];
  return (
    <div className="flex flex-col">
      <Copyright />
      <div className="flex flex-row justify-center">
        <NavigationList items={items} textVariant="body1" />
      </div>
    </div>
  );
};
