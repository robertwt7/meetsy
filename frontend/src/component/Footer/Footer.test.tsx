import { render } from "src/testutils/test-utils";
import { screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Footer } from "./Footer";

jest.mock("next/router", () => ({
  useRouter: jest.fn().mockReturnValue({
    pathname: "/",
  }),
}));

describe("Footer Component", () => {
  it("should match snapshot", () => {
    const { baseElement } = render(<Footer />);
    expect(baseElement).toMatchSnapshot();
  });

  test("should render one of the footer links", () => {
    render(<Footer />);

    expect(screen.getByTestId("privacypolicy")).toBeInTheDocument();
  });
});
