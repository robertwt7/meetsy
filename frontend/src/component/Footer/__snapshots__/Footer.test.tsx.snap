// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should match snapshot 1`] = `
<body>
  <div>
    <div
      class="flex flex-col"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1vl95ex-MuiTypography-root"
      >
        Copyright Â© 
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1kgfegd-MuiTypography-root-MuiLink-root"
          href="http://localhost:3000"
        >
          Meetsy
        </a>
         
        2022
        .
      </p>
      <div
        class="flex flex-row justify-center"
      >
        <ul
          class="flex flex-row items-center justify-center space-x-8 pl-0"
        >
          <li
            class="w-full list-none"
            data-testid="privacypolicy"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-17dcqu8-MuiTypography-root-MuiLink-root"
              data-testid="privacypolicy-a"
              href="/privacy-policy"
              target="_self"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-10gb19z-MuiTypography-root"
              >
                Privacy Policy
              </p>
            </a>
          </li>
          <li
            class="w-full list-none"
            data-testid="termsOfService"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-17dcqu8-MuiTypography-root-MuiLink-root"
              data-testid="termsOfService-a"
              href="/terms-of-service"
              target="_self"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-10gb19z-MuiTypography-root"
              >
                Terms of Service
              </p>
            </a>
          </li>
          <li
            class="w-full list-none"
            data-testid="github"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-zr39bh-MuiTypography-root-MuiLink-root"
              data-testid="github-a"
              href="https://github.com/robertwt7/meetsy"
              target="_self"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-10gb19z-MuiTypography-root"
              >
                Github
              </p>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</body>
`;
